File: src/main/Python/InsertSort.py
class InsertSort:
    def insertSort(self, nums):
        n = len(nums)
        for i in range(n):
            temp = nums[i]
            j = i-1
            while j >= 0 and temp < nums[j]:
                nums[j+1] = nums[j]
                j -= 1
            nums[j+1] = temp
if __name__ == '__main__':
    nums = [49, 38, 65, 97, 76, 13, 27, 49]
    test = InsertSort()
    test.insertSort(nums)
    print(nums)

File: src/main/java/InsertSort.java
/**
 * @author LiuZhiguo
 * @date 2019/10/11 20:14
 */
public class InsertSort {
    void insertSort(int R[], int n){
        int i, j;
        int temp;
        for (i=1;i<n;i++){
            temp = R[i];
            j = i-1;
            while (j >=0 && temp < R[j]){
                R[j+1] = R[j];
                j--;
            }
            R[j+1] = temp;
        }
    }
    public static void main(String[] args){
        InsertSort insertSort = new InsertSort();
        int[] R = {49,38,65,97,76,13,27,49};
        insertSort.insertSort(R, R.length);
        for (int i=0;i<R.length;i++) {
            System.out.println(R[i]);
        }
    }
}
